plankton/core/simulation.py

```diff
@@ -83,6 +86,28 @@ def __init__(self, device, adapter, control_server=None):
         self._device_connected = True
         self._stop_commanded = False
 
+        # Constructing the control server must be deferred until the end,
+        # because the construction is not complete at this point
+        self._control_server = None  # Just initialize to None and use property setter afterwards
+        self.control_server = control_server
+
+    def _create_control_server(self, control_server):
+        if control_server is None or isinstance(control_server, ControlServer):
+            return control_server
+        else:
+            arguments = control_server.split(':')
+
+            if len(arguments) not in (1, 2):
+                raise RuntimeError(
```

Comment:
@MikeHart85 this is one of the errors I meant in my comment in #135. The message is probably okay already, but it needs to be a distinct type so that the message can be displayed in a good way. Tomorrow I'll try to make some progress on #134, where I've already worked on the display part.